MaintenanceManager/MaintenanceManager.cpp

```diff
@@ -2292,36 +2291,32 @@ namespace WPEFramework
                     }
                 }
                 result = true;
+                if(task_stopTimer())
+                {
+                    LOGINFO("Stopped Timer Successfully..");
+                }
+                else
+                {
+                    LOGERR("task_stopTimer() did not stop the Timer...");
+                }
+                task_thread.notify_one();
+                if (m_thread.joinable())
+                {
+                    m_thread.join();
+                    LOGINFO("Thread joined successfully");
+                }
+                if (UNSOLICITED_MAINTENANCE == g_maintenance_type && !g_unsolicited_complete)
```

Comment:
This check must not be moved here. This is the one that prevents unsolicited maintenance from being called twice